lib/protobuf/field/base_field.rb

```diff
@@ -172,16 +172,19 @@ def wire_type
       #
 
       def define_accessor(simple_field_name, fully_qualified_field_name)
-        message_class.class_eval do
-          define_method("#{simple_field_name}!") do
-            @values[fully_qualified_field_name]
+        message_class.class_eval <<-ruby, __FILE__, __LINE__
+          def #{simple_field_name}!
+            @values[:#{fully_qualified_field_name}]
           end
-        end
 
-        message_class.class_eval do
-          define_method(simple_field_name) { self[fully_qualified_field_name] }
-          define_method("#{simple_field_name}=") { |v| self[fully_qualified_field_name] = v }
-        end
+          def #{simple_field_name}
+            self[:#{fully_qualified_field_name}]
+          end
+
+          def #{simple_field_name}=(value)
+            self[:#{fully_qualified_field_name}] = value
```

Comment:
Actually, maybe it should look like `:"#{fully_qualified_field_name}"` to support weird symbols like `:"some.sym.yolo"`.